web-client/src/components/SideNavMenu/SideNavMenu.tsx

```diff
@@ -0,0 +1,130 @@
+import { Menu } from 'antd';
+import React from 'react';
+import { Link, RouteProps } from 'react-router-dom';
+import { COLORS } from 'src/theme/colors';
+import { MenuItem } from 'src/types/menu-item';
+import styled from 'styled-components';
+
+const SideNavMenuItem: React.FC<SideNavMenuItemProps> = ({
+  item,
+  closeSider,
+  ...other
+}) => {
+  const { SubMenu } = Menu;
+  let menuItem: React.ReactNode;
+  if (item.children) {
+    menuItem = (
+      <SubMenu
+        key={item.id}
+        title={
+          <>
+            {item.icon}
+            {item.title}
+          </>
+        }
+        {...other}
+      >
+        {item.children.map((subItem: MenuItem) => (
+          <SideNavMenuItem
+            key={subItem.id}
+            item={subItem}
+            closeSider={closeSider}
+            {...other}
+          />
+        ))}
+      </SubMenu>
+    );
+  } else {
+    menuItem = (
+      <Menu.Item key={item.id} {...other}>
+        {item.icon || null}
+        {item.title}
+      </Menu.Item>
+    );
+  }
+  return (
+    <>
+      {item.location ? (
+        // TODO I couldn't find a proper way here to solve type errors
+        // TODO Location is a JS library
+        // eslint-disable-next-line @typescript-eslint/ban-ts-ignore
+        // @ts-ignore
+        <Link to={{ pathname: item.location.path }} onClick={closeSider}>
+          {menuItem}
+        </Link>
```

Comment:
Any advice here rather than @ts-ignore?

Reply:
What’s the error you’re trying to fix by using `ts-ignore`?

Reply:
@sabind the issue here is that `react-app-location` doesn't have a type definition and so when he specified the type of item.location as `Location`, typescript considers it to be `window.location`.
I fixed this issue by using { path: string } as the type for item.location.